setup/src/installer.ts

```diff
@@ -26,9 +26,9 @@ async function configureOutputs(
   core.setOutput(`${tool}-path`, path);
   core.setOutput(`${tool}-exe`, await which(tool));
   if (tool == 'stack') {
-    const sr =
-      process.env['STACK_ROOT'] ??
-      (os === 'win32' ? 'C:\\sr' : `${process.env.HOME}/.stack`);
+    const stackRootCommand = 'stack path --stack-root';
```

Comment:
_Depending_ on other parameters, this might need option `--system-ghc` (see experiments at #231).
Also, on Windows this stack invocation installs MSYS2.

These points make me hesitant on whether it was a good idea of invoking `stack` here to compute the `stack-root`.  Maybe things were good as they were.